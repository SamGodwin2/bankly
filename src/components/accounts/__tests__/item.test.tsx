import * as React from 'react'
import TestRenderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import { render, screen } from "@testing-library/react";
import AccountItem from '../item';

import type { ReactTestRendererJSON } from 'react-test-renderer';
import type { Account } from '../../../types';

const renderer = ShallowRenderer.createRenderer()

describe('Account item component', () => {

  const mockAccount: Account = {
    account_id: '1234',
    balance: {
      amount: {
        currency: 'EUR',
        value: 2000.50
      }
    }
  }

  test('should render and match the snapshot', () => {
    renderer.render(<AccountItem account={mockAccount}/>)
    const output = TestRenderer.create(<AccountItem account={mockAccount}/>).toJSON()

    expect(output).toMatchSnapshot();
  });

  test('should render as an list item', function () {
    const testRenderer = TestRenderer.create(
      <AccountItem account={mockAccount}/>,
    )
    const output = testRenderer.toJSON() as ReactTestRendererJSON

    expect(output.type).toBe('li')
  })

  test('should render balance correctly', function () {
    render(<AccountItem account={mockAccount}/>)

    expect(screen.getByLabelText(/Total EUR/)).toHaveTextContent('â‚¬2,000.50')
  })
});
