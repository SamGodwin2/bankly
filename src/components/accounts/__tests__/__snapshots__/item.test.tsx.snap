// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account item component should render and match the snapshot 1`] = `
<div
  className="account"
  id="1234"
>
  <div
    className="total"
    id="total-1234"
  >
    Total 
    EUR
  </div>
  <h3
    aria-labelledby="total-1234"
  >
    <strong>
      2000.5
    </strong>
  </h3>
</div>
`;
